### PR TITLE
feat: build using ubuntu and add code-coverage reports

### PR DESCRIPTION
**Pull request by alsami**
* Drop additional TFMs in Test-Project
* Remove conditional import in one of the tests
* Add dependencies to collect code-coverage
* Add basic `codecov.yml`
* Update `build.ps1`, `build.psd1` and `build.psm1`
* Update `.gitignore` & `README.md`

### CODE REVIEW

File: src/Autofac.Integration.AspNetCore.Multitenant/Autofac.Integration.AspNetCore.Multitenant.csproj [hunk 1]
```diff
@@ -13,7 +13,6 @@
     <PackageId>Autofac.AspNetCore.Multitenant</PackageId>
     <PackageTags>autofac;di;ioc;dependencyinjection;aspnet;aspnetcore</PackageTags>
     <PackageReleaseNotes>Release notes are at https://github.com/autofac/Autofac.AspNetCore.Multitenant/releases</PackageReleaseNotes>
-    <PackageIcon>icon.png</PackageIcon>
```
Comment (tillig) [inline src/Autofac.Integration.AspNetCore.Multitenant/Autofac.Integration.AspNetCore.Multitenant.csproj:16]: Why did the package icon disappear?

**Review by tillig — CHANGES_REQUESTED** [review]
Just a question about PackageIcon and we're good here.

File: src/Autofac.Integration.AspNetCore.Multitenant/Autofac.Integration.AspNetCore.Multitenant.csproj (hunk 1 above)
Comment (alsami) [reply src/Autofac.Integration.AspNetCore.Multitenant/Autofac.Integration.AspNetCore.Multitenant.csproj:16]: Good question. Scusi, fixing it quickly

**Review by tillig — APPROVED** [review]
🎉

**Pull request merged**